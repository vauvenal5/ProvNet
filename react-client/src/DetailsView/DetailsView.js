import React from 'react';
import { connect } from 'react-redux';

import { Grid, Header, Label, Icon, Dimmer, Loader, Segment, Button } from 'semantic-ui-react';

import TagsView from '../TagsView';
import ContractForm from "../ContractForm";
import ProvContract from '../models/ProvContract';

export const DetailsView = ({selected, contract}) => {    
    return (
        <Grid padded>
            <Grid.Row>
                <Grid.Column>
                    <Header as="h1" dividing>
                        ProvContract
                        <Dimmer.Dimmable>
                        <Label as="a" tag size="tiny">
                            <Icon name="edit" fitted/>
                            
                            <Label.Detail><Dimmer active={true}>
                                <Loader size="tiny"/>
                            </Dimmer></Label.Detail>
                        </Label>
                        </Dimmer.Dimmable>
                        {/* <Dimmer.Dimmable as={() => <Label as="a" size="tiny"><Icon name="edit" fitted/></Label>} dimmed={true}>
                        
                            
                            <Dimmer active={true}>
                                <Loader size="tiny"/>
                            </Dimmer>
                            </Dimmer.Dimmable> */}
                        {/* <Button size="tiny">
                            <Icon name="edit" fitted/>
                            <Dimmer active={true}>
                                <Loader size="tiny"/>
                            </Dimmer>
                        </Button> */}
                        <Header.Subheader>
                            Contract Address: {selected}
                        </Header.Subheader>
                    </Header>
                </Grid.Column>
            </Grid.Row>
            <Grid.Row columns={2}>
                <Grid.Column stretched>
                    <ContractForm.Component imageUrl={contract.details.logoUrl} description={contract.details.description}/>
                </Grid.Column>
                <Grid.Column floated="right" width={5}>
                    <TagsView.Component 
                        tags={contract.getTags()}
                    />
                </Grid.Column>
            </Grid.Row>
        </Grid>
    );
}

const selectedContract = (contracts, selected) => {
    if(selected === undefined) {
        return new ProvContract(selected);
    }
    return contracts[selected];
}

//container part
export const mapStateToProps = (state) => {
    let selected = state.contracts.selected[0];

    return {
        selected: selected,
        contract: selectedContract(state.contracts, selected),
    };
}

export default connect(mapStateToProps)(DetailsView);