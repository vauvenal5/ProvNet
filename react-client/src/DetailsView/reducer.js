import * as modelActions from "../modelActions";

const detailsViewReducer = (state={}, action) => {
    switch(action.type) {
        case modelActions.types.contractSelect:
            return Object.assign({}, state, {
                selected: action.address
            });
        default:
            return state;
    }
}

export default detailsViewReducer;