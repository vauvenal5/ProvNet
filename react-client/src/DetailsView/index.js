import Component from "./DetailsView";
import reducer from "./reducer";

export default {Component, reducer};