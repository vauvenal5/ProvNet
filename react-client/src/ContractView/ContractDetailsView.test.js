import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import testContract from "../models/ProvContract.testHelper";
import DetailsView from './ContractDetails';

Enzyme.configure({ adapter: new Adapter() });

const setup = (propOverrides) => {
    const props = Object.assign({}, testContract.details, propOverrides);

    const wrapper = shallow(<DetailsView {...props}/>);

    return {
        props,
        wrapper,
        render: wrapper.debug()
    }
};

describe("DetailsView", () => {Component
    it("should render empty URL", () => {
        const { render } = setup({logoUrl: ""});
        expect(render).toMatchSnapshot();
    });
});

describe("DetailsView", () => {
    it("should render with custom URL", () => {
        const { render } = setup();
        expect(render).toMatchSnapshot();
    });
});