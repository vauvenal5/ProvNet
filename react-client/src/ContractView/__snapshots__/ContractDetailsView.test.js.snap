// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractForm should render empty URL 1`] = `
"<ItemGroup>
  <Item>
    <ItemImage src=\\"https://react.semantic-ui.com/images/wireframe/square-image.png\\" size=\\"small\\" circular={true} />
    <ItemContent>
      <ItemHeader>
        Description
      </ItemHeader>
      <ItemDescription>
        This is a test contract!
      </ItemDescription>
    </ItemContent>
  </Item>
</ItemGroup>"
`;

exports[`ContractForm should render with custom URL 1`] = `
"<ItemGroup>
  <Item>
    <ItemImage src=\\"https://react.semantic-ui.com/images/wireframe/square-image.png\\" size=\\"small\\" circular={true} />
    <ItemContent>
      <ItemHeader>
        Description
      </ItemHeader>
      <ItemDescription>
        This is a test contract!
      </ItemDescription>
    </ItemContent>
  </Item>
</ItemGroup>"
`;
